File: alura/carreira_java/example19/src/br/com/flaviogf/Account.java
package br.com.flaviogf;

public class Account {
    double amount;
    String number;
    String agency;
    String holder;

    public void deposit(double value) {
        this.amount += value;
    }

    public boolean withdraw(double value) {
        if (this.amount - value >= 0) {
            this.amount -= value;
            return true;
        }

        return false;
    }
}

File: alura/go_oo/005/bank/accounts/account.go
package accounts

import (
	"errors"

	"github.com/flaviogf/bank/customers"
)

type Account struct {
	Holder  customers.Customer
	Agency  string
	Number  string
	balance int
}

func (self *Account) Deposit(value int) error {
	if value <= 0 {
		return errors.New("value to deposit must be positive")
	}

	self.balance += value

	return nil
}

func (self *Account) Balance() int {
	return self.balance
}
